lib/charty/table_adapters/datasets_adapter.rb

```diff
@@ -0,0 +1,54 @@
+module Charty
+  module TableAdapters
+    class DatasetsAdapter
+      extend Forwardable
+      include Enumerable
+
+      def self.make(dataset)
+        GenericDatasetsAdapter.new(dataset)
+      end
+
+      def self.supported?(data)
+        defined?(Datasets::Dataset) && data.is_a?(Datasets::Dataset)
+      end
+
+      def initialize(dataset)
+        @dataset = dataset
+        @records = []
+      end
+
+      def each(&block)
+        return to_enum(__method__) unless block_given?
+
+        fetch_all if @records.empty?
+        @records.each(&block)
+      end
+
+      private def fetch_all
+        @dataset.each do |record|
+          @records << record
+        end
+      end
+
+      def column(i)
+        fetch_all if @records.empty?
+        @records.map {|record| record[i] }
+      end
+
+      # @param [Integer] i  Row index
+      # @param [Symbol,String,Integer] j  Column index
+      def [](i, j)
+        fetch_all if @records.empty?
+        @records[i][j]
+      end
+
+      class GenericDatasetsAdapter < self
```

Comment:
I think I don't understand the need for "GenericDatasetsAdapter class" at this point. In what cases is it effective?

Reply:
I introduced this adapter for subclasses of Datasets which have their own Record class.
I've renamed it to RecordCollectionAdapter.

Reply:
The reason why I introduce this class as a subclass of DatasetsAdapter is that I want to introduce adapters for MNIST, CIFAR, libsvm, etc. in the future.